FE-2950 Add geographical map using OpenLayers (#FE-2950)

JeroenGommans opened the pull request:
- Added geographical map using OpenLayers
- new card fields: wktGeometry and wktProjection
- new config options in web-ui.json: "geomap"

Signed-off-by: Jeroen Gommans <jeroen@anjersoftwaredesign.nl>

freddidierRTE commented on ui/main/src/app/modules/feed/components/map/map.component.ts:
```diff
@@ -0,0 +1,221 @@
+            displayFeatureInfo(evt);
+        });
+
+        function displayFeatureInfo(evt) {
```
To be more meaningfull , i would name it displayLightCardIfNecessary

JeroenGommans replied:
Done

freddidierRTE commented:
1 - When we  do not use the feature we have a blue cross on the screen (enableMap :false)  , see screenshot : 

![image](https://user-images.githubusercontent.com/58251627/173299112-5f8bdcb6-240d-49f5-bdf1-1821bea66492.png)


2- When opening opfab or closing a card detail , the map component is zooming progressively , could it be done instantly instead of progressively ? 

3-  Height of the map depends on the number of cards in the feed : when the feed it "full"  the map takes all available space but if the feed is empty or with a few cards , there's some place left on the bottom of the screen . Height of the map shall as well adapt when hiding the timeline (this is the case when the feed is "full")  

4 - Add copyright header in all new files (html ,scss and ts ) 

5 - To be done in another issue : 
- deal with the case where there is more than on card at the same geographical  position 
- night/day mode 
- testing with cypress

JeroenGommans commented on ui/main/src/app/modules/share/light-card/light-card.component.ts:
```diff
@@ -38,10 +38,10 @@ export class LightCardComponent implements OnInit, OnDestroy {
     @Input() public lightCard: LightCard;
     @Input() public displayUnreadIcon = true;
     @Input() displayContext: any = DisplayContext.REALTIME;
+    @Input() addSpacer = false;
```
Done

JeroenGommans commented on ui/main/src/app/modules/feed/components/map/map.component.scss:
```diff
@@ -0,0 +1,30 @@
+.of-map {
```
Done

JeroenGommans commented on ui/main/src/app/modules/feed/components/map/map.component.ts:
```diff
@@ -0,0 +1,221 @@
+
+        lightCards
+            .filter(lc => lc.wktGeometry)
+            .forEach(
```
Done

JeroenGommans commented on ui/main/src/app/modules/feed/components/map/map.component.scss:
```diff
@@ -0,0 +1,30 @@
+// Map at filter level
+//  margin-top: 10px;
+//  height: calc(100% - 10px);
+
```
Done

JeroenGommans commented:
> 1 - When we do not use the feature we have a blue cross on the screen (enableMap :false) , see screenshot :

Fixed
 
> 2- When opening opfab or closing a card detail , the map component is zooming progressively , could it be done instantly instead of progressively ?

I added a new config option "feed.geomap.zoomDuration". If set to 0 the animation is disabled
 
> 3- Height of the map depends on the number of cards in the feed : when the feed it "full" the map takes all available space but if the feed is empty or with a few cards , there's some place left on the bottom of the screen . Height of the map shall as well adapt when hiding the timeline (this is the case when the feed is "full")

I'm looking into this one.
> 4 - Add copyright header in all new files (html ,scss and ts )

Done

> 5 - To be done in another issue :
> 
>     * deal with the case where there is more than on card at the same geographical  position

It is on our backlog. I'll add an GitHub issue as well.
 
>     * night/day mode

This should already work..

>     * testing with cypress
I'll add an GitHub issue and put it on our backlog

JeroenGommans commented:
> > 3- Height of the map depends on the number of cards in the feed : when the feed it "full" the map takes all available space but if the feed is empty or with a few cards , there's some place left on the bottom of the screen . Height of the map shall as well adapt when hiding the timeline (this is the case when the feed is "full")
> 
> I'm looking into this one.
> 

Fixed

freddidierRTE commented:
When enableMap set to false , i have an exception in ui : 
![image](https://user-images.githubusercontent.com/58251627/173873450-27aa4a1d-9666-482d-b2e0-fac7984babd9.png)
